### PR TITLE
minio 20160713214605

### PR DESCRIPTION
**Pull request by nl5887**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/minio.rb [hunk 1]
```diff
@@ -1,18 +1,13 @@
+require "language/go"
```
Comment (DomT4) [inline Formula/minio.rb]: Don't need this unless you're using `go_resource`s.

File: Formula/minio.rb [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
   homepage "https://github.com/minio/minio"
-  url "https://github.com/minio/minio/archive/RELEASE.2016-06-03T19-32-05Z.tar.gz"
-  version "20160603193205"
-  sha256 "8ecf52cc344c99f84bef632b8dfecf8fdc6a8493a541b2fe42f5b22d8bbfa802"
+  url "https://github.com/minio/minio/archive/RELEASE.2016-07-13T21-46-05Z.tar.gz"
+  version "20160713214605"
+  sha256 "40cd2faf5dd9503c20d2f5a5ad67277e960561fca5869a9d724d1efa763021df"
   head "https://github.com/minio/minio.git"
 
-  bottle do
```
Comment (DomT4) [inline Formula/minio.rb]: Don't need to delete this block on updates. CI knows how to handle it.

File: Formula/minio.rb [hunk 3]
```diff
@@ -21,12 +16,14 @@ def install
...
+    cd clipath do
+      system "go", "build", "-o", buildpath/"minio"
     end
+
+    bin.install buildpath/"minio"
   end
 
   test do
-    system "#{bin}/minio"
+    system "#{bin}/minio", "version"
```
Comment (DomT4) [inline Formula/minio.rb]: If we're getting a version output here might be worth making this:
```ruby
assert_match version.to_s, shell_output("#{bin}/minio version")
```

File: Formula/minio.rb (hunk 1 above)
Comment (DomT4) [inline Formula/minio.rb]: This remains an issue.

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: Formula/minio.rb [hunk 4]
```diff
@@ -1,17 +1,11 @@
 class Minio < Formula
   desc "object storage server compatible with Amazon S3"
   homepage "https://github.com/minio/minio"
-  url "https://github.com/minio/minio/archive/RELEASE.2016-06-03T19-32-05Z.tar.gz"
-  version "20160603193205"
-  sha256 "8ecf52cc344c99f84bef632b8dfecf8fdc6a8493a541b2fe42f5b22d8bbfa802"
-  head "https://github.com/minio/minio.git"
+  url "https://github.com/minio/minio/archive/RELEASE.2016-07-13T21-46-05Z.tar.gz"
+  version "2016-07-13T21:46:05Z"
```
Comment (dunn) [inline Formula/minio.rb]: This is interpreted as an earlier version than `20160603193205`, so `brew upgrade` will skip it.  Can you stick with the original version schema?

Comment (nl5887) [inline Formula/minio.rb]: Hmm, sorry about that. I've updated the formula.

Comment (MikeMcQuaid) [inline Formula/minio.rb]: Don't see your updates yet?

File: Formula/minio.rb [hunk 5]
```diff
@@ -1,17 +1,11 @@
...
+  sha256 "40cd2faf5dd9503c20d2f5a5ad67277e960561fca5869a9d724d1efa763021df"
 
-  bottle do
-    cellar :any_skip_relocation
-    sha256 "29d5d319618c235dc75b86b9849b1b8f9d3ccc8c5b313c28e73167119ae43c4e" => :el_capitan
-    sha256 "431a20fe6725e32f3a576937815b28e7af8fecfa1faab29e3c1d24d97edb0941" => :yosemite
-    sha256 "c79ee7ccfea2b00db2b8792a479bac3142ce9567c7c69eade8a435b99600a77b" => :mavericks
-  end
```
Comment (MikeMcQuaid) [inline Formula/minio.rb]: Leave the bottle block as-is; we'll update when we pull.

File: Formula/minio.rb [hunk 6]
```diff
@@ -21,12 +15,21 @@ def install
...
     clipath.install Dir["*"]
 
-    cd buildpath/"src/github.com/minio/minio/" do
-      system "go", "build", "-o", bin/"minio", "-v"
+    cd clipath do
+      if build.head?
+        system "go", "build", "-o", buildpath/"minio"
+      else
+        release = "RELEASE.2016-07-13T21-46-05Z"
+        commit = "3f27734c22212f224037a223439a425e6d2b653a"
```
Comment (MikeMcQuaid) [inline Formula/minio.rb]: This shouldn't be hardcoded; consider using a Git tag instead and reading this from the buildsystem.

Comment (nl5887) [inline Formula/minio.rb]: having some issues retrieving the current tag from the buildsystem. Trying to catch the output of `"git", "tag", "--points-at", "HEAD"` and save it into variable, but pipe_output / shell_output are not available. Any suggestions?

Comment (MikeMcQuaid) [inline Formula/minio.rb]: Use backticks outside tests.

File: Formula/minio.rb [hunk 7]
```diff
@@ -21,12 +19,24 @@ def install
...
 
-    cd buildpath/"src/github.com/minio/minio/" do
-      system "go", "build", "-o", bin/"minio", "-v"
+    cd clipath do
+      if build.head?
+        system "go", "build", "-o", buildpath/"minio"
+      else
+        minio_release = `git tag --points-at HEAD`
+        version = minio_release.gsub(/RELEASE\./, "")
+        minio_version = version
```
Comment (MikeMcQuaid) [inline Formula/minio.rb]:
```ruby
minio_version = `git tag --points-at HEAD`.gsub(/RELEASE\./, "").chomp
```

File: Formula/minio.rb [hunk 8]
```diff
@@ -21,12 +19,24 @@ def install
...
+        minio_commit = `git rev-parse HEAD`
```
Comment (MikeMcQuaid) [inline Formula/minio.rb]:
```ruby
minio_commit = `git rev-parse HEAD`.chomp
```

File: Formula/minio.rb [hunk 9]
```diff
@@ -21,12 +19,24 @@ def install
...
+      end
     end
+
+    bin.install buildpath/"minio"
   end
 
   test do
-    system "#{bin}/minio"
+    minio_version = DateTime.parse(version).strftime("%Y-%m-%dT%H-%M-%SZ")
+    assert_match minio_version, shell_output("#{bin}/minio version", "version doesn't match")
```
Comment (MikeMcQuaid) [inline Formula/minio.rb]: Second argument for `shell_output` should be left blank: https://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/5982/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_minio/

File: Formula/minio.rb [hunk 10]
```diff
@@ -21,12 +19,23 @@ def install
...
+        system "go", "build", "-ldflags", "-X main.minioVersion=#{minio_version} -X main.minioReleaseTag=#{minio_release} -X main.minioCommitID=#{minio_commit}", "-o", buildpath/"minio"
+      end
     end
+
+    bin.install buildpath/"minio"
   end
 
   test do
-    system "#{bin}/minio"
+    minio_version = DateTime.parse(version).strftime("%Y-%m-%dT%H-%M-%SZ")
```
Comment (MikeMcQuaid) [inline Formula/minio.rb]: Unfortunately 1.8 (which we still have to support for now) doesn't support `.parse` here so better maybe to just use the old, crappy test: https://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/5998/version=mavericks/testReport/junit/brew-test-bot/mavericks/test_minio/

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**